gohelper.go

```diff
@@ -11,89 +11,347 @@ import (
-	p.DocRoot = os.Getenv("PLATFORM_DOCUMENT_ROOT")
-	p.TreeId = os.Getenv("PLATFORM_TREE_ID")
-	p.Branch = os.Getenv("PLATFORM_BRANCH")
-	p.Environment = os.Getenv("PLATFORM_ENVIRONMENT")
-	p.Project = os.Getenv("PLATFORM_PROJECT")
-	p.Entropy = os.Getenv("PLATFORM_PROJECT_ENTROPY")
-	p.Socket = os.Getenv("SOCKET")
-	p.Port = os.Getenv("PORT")
+	p.application = parsedApplication
 
 	return p, nil
 }
 
+// This function returns a new PlatformConfig object, representing
+// the abstracted Platform.sh environment.  If run on not a Platform.sh
+// environment (eg, a local computer) then it will return nil and an error.
+func NewConfig() (*PlatformConfig, error) {
+	return NewConfigReal(os.Getenv, "PLATFORM_")
+}
+
+// Checks whether the code is running in a build environment.
+func (p *PlatformConfig) InBuild() bool {
+	return p.environment == ""
+}
+
+// Checks whether the code is running in a runtime environment.
+func (p *PlatformConfig) InRuntime() bool {
+	return p.environment != ""
+}
+
+// Determines if the current environment is a Platform.sh Enterprise environment.
+func (p *PlatformConfig) OnEnterprise() bool {
+	return p.mode == "enterprise"
+}
+
+// Determines if the current environment is a production environment.
+//
+// Note: There may be a few edge cases where this is not entirely correct on Enterprise,
+// if the production branch is not named `production`.  In that case you'll need to use
+// your own logic.
+func (p *PlatformConfig) OnProduction() bool {
+	if !p.InRuntime() {
+		return false
+	}
+
+	var prodBranch string
+	if p.OnEnterprise() {
+		prodBranch = "production"
+	} else {
+		prodBranch = "master"
+	}
+
+	return p.branch == prodBranch
+}
+
+// The name of the application, as defined in its configuration.
+func (p *PlatformConfig) ApplicationName() string {
+	return p.applicationName
+}
+
+// An ID identifying the application tree before it was built: a unique hash
+// is generated based on the contents of the application's files in the
+// repository.
+func (p *PlatformConfig) TreeId() string {
+	return p.treeId
+}
+
+// The absolute path to the application.
+func (p *PlatformConfig) AppDir() string {
+	return p.appDir
+}
+
+// The project ID.
+func (p *PlatformConfig) Project() string {
+	return p.project
+}
+
+// A random string generated for each project, useful for generating hash keys.
+func (p *PlatformConfig) Entropy() string {
+	return p.entropy
+}
+
+// The Git branch name.
+func (p *PlatformConfig) Branch() string {
+	return p.branch
+}
+
+// The environment ID (usually the Git branch plus a hash).
+func (p *PlatformConfig) Environment() string {
+	return p.environment
+}
+
+// The absolute path to the web root of the application.
+func (p *PlatformConfig) DocumentRoot() string {
+	return p.documentRoot
+}
+
+// The hostname of the Platform.sh default SMTP server (an empty string if
+// emails are disabled on the environment).
+func (p *PlatformConfig) SmtpHost() string {
```

Comment:
This should be `SMTPHost`, no?

Reply:
I've seen this sort of camelcasing more commonly than SMTPHost style.  Although that may also be language-dependent.  What's the Go standard here?